pygmt/tests/test_grdtrack.py

```diff
@@ -0,0 +1,134 @@
+"""
+Tests for grdtrack
+"""
+import os
+
+import pandas as pd
+import pytest
+
+from .. import grdtrack
+from .. import which
+from ..datasets import load_earth_relief, load_ocean_ridge_points
+from ..exceptions import GMTInvalidInput
+from ..helpers import data_kind
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+TEMP_TRACK = os.path.join(TEST_DATA_DIR, "tmp_track.txt")
+
+
+def test_grdtrack_input_dataframe_and_dataarray():
+    """
+    Run grdtrack by passing in a pandas.DataFrame and xarray.DataArray as inputs
+    """
+    dataframe = load_ocean_ridge_points()
+    dataarray = load_earth_relief().sel(lat=slice(-49, -42), lon=slice(-118, -107))
+
+    output = grdtrack(points=dataframe, grid=dataarray, newcolname="bathymetry")
+    assert isinstance(output, pd.DataFrame)
+    assert output.columns.to_list() == ["longitude", "latitude", "bathymetry"]
+    assert output.iloc[0].to_list() == [-110.9536, -42.2489, -2823.96637605]
```

Comment:
```suggestion
    npt.assert_allclose(output.iloc[0], [-110.9536, -42.2489, -2823.96637605])
```

When checking floating point numbers, always use `numpy.testing.assert_allclose` or something like it. The equality will fail depending on the system and rounding errors. The `numpy.testing` module has a lot of useful function for testing arrays.

Reply:
Interesting point, I'll definitely keep this in mind! I've modified this line as requested as well as 3 other similar lines in `test_grdtrack.py`.